src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -434,6 +438,44 @@ private IOperation CreateBoundCallOperation(BoundCall boundCall)
             return new CSharpLazyInvocationOperation(this, boundCall, targetMethod.GetPublicSymbol(), isVirtual, _semanticModel, syntax, type, constantValue, isImplicit);
         }
 
+        private IOperation CreateBoundFunctionPointerInvocationOperation(BoundFunctionPointerInvocation boundFunctionPointerInvocation)
+        {
+            if (boundFunctionPointerInvocation.HasErrors)
+            {
+                return new CSharpLazyInvalidOperation(
+                    this,
+                    boundFunctionPointerInvocation,
+                    _semanticModel,
+                    boundFunctionPointerInvocation.Syntax,
+                    boundFunctionPointerInvocation.Type.GetPublicSymbol(),
+                    ConvertToOptional(boundFunctionPointerInvocation.ConstantValue),
+                    isImplicit: boundFunctionPointerInvocation.WasCompilerGenerated);
+            }
+
+            return new CSharpLazyInvocationOperation(
+                this,
+                boundFunctionPointerInvocation,
+                boundFunctionPointerInvocation.FunctionPointer.Signature.GetPublicSymbol(),
```

Comment:
>boundFunctionPointerInvocation.FunctionPointer.Signature.GetPublicSymbol() [](start = 16, length = 74)

This looks suspicious. I am not sure it is the right thing to pretend that this is the method that is getting invoked.

Reply:
I would say we need a special IOperation node to represent a function pointer invocation.

---
In reply to: [428789347](https://github.com/dotnet/roslyn/pull/44436#discussion_r428789347) [](ancestors = 428789347)

Reply:
I don't think I agree. Function pointer invocations are one of the `callX` Il instructions, and I feel like it should be represented as a standard invocation. Semantically, it uses the same parameter conversion rules as a standard invocation, the same invocation order rules... I'm also having trouble seeing when you'd want to analyze invocations and not also look for function pointer invocations. @cyrusnajmabadi, do you want to weigh as well here?

---
In reply to: [428790455](https://github.com/dotnet/roslyn/pull/44436#discussion_r428790455) [](ancestors = 428790455,428789347)

Reply:
> I'm also having trouble seeing when you'd want to analyze invocations and not also look for function pointer invocations.

I don't believe this is relevant. We don't bundle things together just because people might want to look at them.

---
In reply to: [428829053](https://github.com/dotnet/roslyn/pull/44436#discussion_r428829053) [](ancestors = 428829053,428790455,428789347)

Reply:
I believe the two first properties are not applicable for function pointer invocations. The values that we are going to return are going to be misleading and simply will not match the API design. FP signature is not a method definition and it is not a member of the pointer instance.
```
    public interface IInvocationOperation : IOperation
    {
        /// <summary>
        /// Method to be invoked.
        /// </summary>
        IMethodSymbol TargetMethod { get; }
        /// <summary>
        /// 'This' or 'Me' instance to be supplied to the method, or null if the method is static.
        /// </summary>
        IOperation Instance { get; }
```

---
In reply to: [428832354](https://github.com/dotnet/roslyn/pull/44436#discussion_r428832354) [](ancestors = 428832354,428829053,428790455,428789347)

Reply:
The third property is also not applicable and will be misleading no matter what value we return:
```
        /// <summary>
        /// True if the invocation uses a virtual mechanism, and false otherwise.
        /// </summary>
        bool IsVirtual { get; }
```

---
In reply to: [428837184](https://github.com/dotnet/roslyn/pull/44436#discussion_r428837184) [](ancestors = 428837184,428832354,428829053,428790455,428789347)

Reply:
I'm going to make this a `NoneOperation` for now, and we can figure out a type shape later.

---
In reply to: [428841690](https://github.com/dotnet/roslyn/pull/44436#discussion_r428841690) [](ancestors = 428841690,428837184,428832354,428829053,428790455,428789347)